### PR TITLE
feat: create new connection on connection validation error in postgres

### PR DESCRIPTION
**Pull request by saxenakshitiz**

### CODE REVIEW

File: query-service-impl/src/main/java/org/hypertrace/core/query/service/postgres/PostgresClientFactory.java [hunk 1]
```diff
@@ -62,57 +59,98 @@ public PostgresClient getPostgresClient(String clusterName) {
...
+      }
     }
 
-    public Connection getConnection() throws SQLException {
-      return dataSource.getConnection();
+    private synchronized void newConnection() throws SQLException {
+      int attempts = 0;
+      while (attempts < maxConnectionAttempts) {
+        try {
+          ++count;
```
Comment (surajpuvvada) [inline query-service-impl/src/main/java/org/hypertrace/core/query/service/postgres/PostgresClientFactory.java:118]: what is count used for ?

Comment (kotharironak) [inline query-service-impl/src/main/java/org/hypertrace/core/query/service/postgres/PostgresClientFactory.java:118]: Same comment on as the previous PR.

**Review by kotharironak — COMMENTED** [review]
lgtm

File: query-service-impl/src/main/java/org/hypertrace/core/query/service/postgres/PostgresClientFactory.java (hunk 1 above)
Comment (saxenakshitiz) [reply query-service-impl/src/main/java/org/hypertrace/core/query/service/postgres/PostgresClientFactory.java]: it is sort of id for connection

Comment (saxenakshitiz) [reply query-service-impl/src/main/java/org/hypertrace/core/query/service/postgres/PostgresClientFactory.java]: done

**Review by kotharironak — APPROVED** [review]

**Pull request merged**